frontend/libs/studio-components/src/components/StudioRecommendedNextAction/context/StudioRecommendedNextActionContext.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { createContext, useMemo, useState } from 'react';
+
+export type StudioRecommendedNextActionContext = {
+  shouldDisplayAction: (actionId: string) => boolean;
+  addAction: (actionId: string) => void;
+  removeAction: (actionId: string) => void;
+};
+
+export const StudioRecommendedNextActionContext =
+  createContext<StudioRecommendedNextActionContext>(undefined);
+
+export type StudioRecommendedNextActionContextProviderProps = {
+  children: React.ReactNode;
+};
+
+export const StudioRecommendedNextActionContextProvider = ({
+  children,
+}: StudioRecommendedNextActionContextProviderProps) => {
+  const [actions, setActions] = useState<Set<string>>(new Set());
+
+  const addAction = (actionId: string): void => {
+    setActions((prevActions) => new Set(prevActions).add(actionId));
+  };
+
+  const shouldDisplayAction = (actionId: string): boolean => {
+    return actions.has(actionId);
+  };
+
+  const removeAction = (actionId: string): void => {
+    setActions((prevActions) => {
+      const newActions = new Set(prevActions);
+      newActions.delete(actionId);
+      return newActions;
+    });
+  };
+
+  const value = useMemo(
+    () => ({
+      shouldDisplayAction,
+      addAction,
+      removeAction,
+    }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I might be wrong in my reasoning behind this memoization, but the idea was that unless `actions` does not change, we do not need to recreate the context value. As this will cause rerenders of all components depending on this context.
But i did some more testing and the memoization seems to have zero effect on rerenders in this case. 
So i'll remove it unless you have any arguments for it's reason to stay?